app/controllers/RemoveNiphlController.scala

```diff
@@ -71,6 +71,7 @@ class RemoveNiphlController @Inject() (
                   sessionRepository.set(answers).map { _ =>
                     Redirect(navigator.nextPage(RemoveNiphlPage, NormalMode, answers))
                   }
+                case _                => Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad().url))
```

Comment:
is this a JR choice according to what @oliverslade said earlier or should we let the global error handler handle it?

Reply:
We still need to have something locally to handle the incomplete match otherwise Scala will complain that there is missing cases

Reply:
Hmmm can we ask oil about it? It might be a case that we can just hide these warnings

Reply:
@besscerule @Reece-Carruthers can you briefly explain what the scenario is where we are using JourneyRecovery?

Reply:
@oliverslade basically to solve some warning issues with incomplete matches I decided to add a case which will catch any failures and redirect the user to the journey recovery

The scenario in question is what would we do in the event of user answers failing to set an answer 

```
if (value) {
              request.userAnswers.set(RemoveNiphlPage, value) match {
                case Success(answers) =>
                  sessionRepository.set(answers).map { _ =>
                    Redirect(navigator.nextPage(RemoveNiphlPage, NormalMode, answers))
                  }
                case _                => Future.successful(Redirect(routes.JourneyRecoveryController.onPageLoad().url))
              }

Reply:
in this case when a user submits and answer to this page (which is whether you want to remove a niphl number (which is a filed that is associated with the user's profile) we set the userAnswers (standard pattern from the scaffold) to true or false for that field. If we set it successfully (which is guess is the most likely option) everything is good - if we don't we now go to the journey recovery page instead of the global error handler. Whether or not we set it successfully depends on if it is not a JSObject I think. So I'm not sure if that counts as a place to use Journey Recovery. Because if the user keeps trying this again it will break because they can only do true or false so if it errors it means we have a bug not that the user is wrong. I guess what I am saying is this doesn't feel like a path the user should ever even be on because if they get here something is broken.

Reply:
Taking a look, theres no hard and fast rule on this. Quite often its used as a fall back for many/any/some/all errors. And some teams use it differently, sometimes you know whats gone wrong and its recoverable, sometimes you dont. Something you can do is route the user back to the start/home page/tasklist or whatever and hope they resolve whatever has gone wrong themselves. OR clear userAnswers at that point (telling the user/giving them the option) for something that might be unresolvable.
I think what you have done here is fine. The question seems like it is a yes/no form so if the value they set is something different, it must mean that something has gone wrong or theyve manipulated the request.
To add to what I mentioned yesterday. If we get to the end of the journey and validated the full user answers into a payload before we send it, and notice the niphl number boolean was not set, we know that the validation will fail, in which case the submission wont happen and we know why. I think in this case its better to go to journey recovery here so the user can potentially resolve the issue. Where as the global error handler error page is a dead end of the user.